pkg/providers/instancetype/instancetype.go

```diff
@@ -112,8 +113,40 @@ func (p *Provider) List(ctx context.Context, kc *corev1beta1.KubeletConfiguratio
 	// Compute fully initialized instance types hash key
 	subnetHash, _ := hashstructure.Hash(subnets, hashstructure.FormatV2, &hashstructure.HashOptions{SlicesAsSets: true})
 	kcHash, _ := hashstructure.Hash(kc, hashstructure.FormatV2, &hashstructure.HashOptions{SlicesAsSets: true})
+	// TODO: remove kubeReservedHash and systemReservedHash one string is once the fields get hashed as a string in kubeletConfiguration
+	// https://github.com/kubernetes-sigs/karpenter/issues/1080
+	kubeReservedHash, systemReservedHash := uint64(0), uint64(0)
+	if kc != nil {
+		kubeReservedSlice := lo.MapToSlice(resources.StringMap(kc.KubeReserved), func(k string, v string) string { return fmt.Sprintf("%s_%s", k, v) })
```

Comment:
Slice isn't a good idea, this could change ordering based on what the map gives us on a given iteration. We should convert the values to a string in the value, but leave them in a map so that it can be still hashed as a set